sdk/go/common/workspace/plugins.go

```diff
@@ -615,24 +615,12 @@ func (spec PluginSpec) Dir() string {
 	return dir
 }
 
-// File gets the expected filename for this plugin.
+// File gets the expected filename for this plugin, excluding any platform specific suffixes (e.g. ".exe" on
+// windows).
 func (spec PluginSpec) File() string {
-	return spec.FilePrefix() + spec.FileSuffix()
-}
-
-// FilePrefix gets the expected default file prefix for the plugin.
-func (spec PluginSpec) FilePrefix() string {
```

Comment:
I highly doubt anyone outside of this repo is using these methods, so I'm OK with deleting them, but just pointing out that technically this _could_ break someone.

From a [search on GH](https://cs.github.com/?scopeName=All+repos&scope=&q=pulumi%2Fsdk%2Fv3%2Fgo%2Fcommon%2Fworkspace#), there are a some uses of `github.com/pulumi/pulumi/sdk/v3/go/common/workspace` outside this repo (e.g. in https://github.com/jaxxstorm/ploy and https://github.com/pulumi/pulumi-terraform-bridge), but they aren't using `PluginSpec`.

Reply:
Yup we've been trying to be pragmatic about "go/common", a lot of the code is pretty much just internal engine code that a few internal tools also happen to depend on (mostly ends up being tfbridge) and so we've been changing it and just taking to hit to go and update those bits as well, rather than being limited in what changes we can make.

I'm also looking at a long term goal of removing _everything_ in common. Stuff that's really shared isn't just shared between the engine and the go SDK but also every other SDK (things like property values) and so we'll generate them from a truly common definition (not sure exactly what that will be, maybe just new grpc defs, maybe json schema, maybe something in pulumi schema). Parts of common just used by the engine will be moved to the engine, and we'll work out how to decouple things like tfbridge/yaml from having to know about internal engine details like how to lookup and start plugins.